code/__HELPERS/patterns.dm

```diff
@@ -71,6 +71,9 @@
 					continue
 				if (AdjT in results) // Ignore existing turfs
 					continue
+				if (barricade_block)
+					if (locate(/obj/structure/barricade) in AdjT)
+						continue
```

Comment:
put this in DirBlocked, dont use locate, check for density, remove barricade_block

Reply:
this would mess with flame grenades and other stuff If I didnt use a special variable

Reply:
did your thing